node-fs-readfile/cat.js

```diff
@@ -3,16 +3,17 @@ import { readFile } from 'node:fs/promises';
 const fileArray = process.argv.slice(2);
 
 async function print() {
-  await Promise.all(
-    fileArray.map(async (file) => {
-      try {
-        const data = await readFile(`./${file}`, { encoding: 'utf8' });
-        console.log(data);
-      } catch (error) {
-        console.log(error.message);
-      }
-    })
-  );
+  const array = await fileArray.map((file) => {
```

Comment:
In this line, you are `await`ing the `fileArray.map` call, which is synchronous. It may look to you like you are `await`ing the readFile call below, but that's not what the syntax says. Take a look at the code and carefully read the syntax:

- Pass an arrow function to fileArray.map
- await the fileArray.map call
- When .map calls the arrow function, execute the code at lines 7 through 12

Since there is no await inside the arrow function, there is nothing happening asynchronously, so there is nothing to await.

The best way to think about `await` is that you use it _only_ on functions that return promises. Array.map returns an array, not a Promise, so you can't `await` it.